internal/setup/gatewaysecret.go

```diff
@@ -0,0 +1,44 @@
+package setup
+
+import (
+	"os"
+
+	certmanagerv1 "github.com/cert-manager/cert-manager/pkg/apis/certmanager/v1"
+	gcertv1alpha1 "github.com/gardener/cert-management/pkg/apis/cert/v1alpha1"
+	"github.com/go-logr/logr"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	"github.com/kyma-project/lifecycle-manager/internal/common"
+	gatewaysecretclient "github.com/kyma-project/lifecycle-manager/internal/gatewaysecret/client"
+	"github.com/kyma-project/lifecycle-manager/internal/pkg/flags"
+	"github.com/kyma-project/lifecycle-manager/pkg/watcher/certificate"
+)
+
+//nolint:ireturn // chosen implementation shall be abstracted
+func SetupCertInterface(kcpClient client.Client,
+	flagVar *flags.FlagVar,
+	setupLog logr.Logger,
+) gatewaysecretclient.CertificateInterface {
```

Comment:
I am not super happy with how we set things up overall. I would like to go forward with this as a pragmatic approach but we need to get clarity on the topic: https://github.com/kyma-project/lifecycle-manager/pull/2406